pkg/provider/azure_vmssflex_cache.go

```diff
@@ -0,0 +1,315 @@
+					klog.V(2).Infof("Skip Uniform VMSS: (%s)", *scaleSet.ID)
+					continue
+				}
+
+				localCache.Store(*scaleSet.ID, &scaleSet)
+			}
+		}
+
+		return localCache, nil
+	}
+
+	if fs.Config.VmssFlexCacheTTLDefaultInSeconds == 0 {
+		fs.Config.VmssFlexCacheTTLDefaultInSeconds = consts.VmssFlexCacheTTLDefaultInSeconds
+	}
+	return azcache.NewTimedcache(time.Duration(fs.Config.VmssFlexCacheTTLDefaultInSeconds)*time.Second, getter)
+}
+
+func (fs *FlexScaleSet) newVmssFlexVMCache() (*azcache.TimedCache, error) {
+	getter := func(key string) (interface{}, error) {
+		klog.V(2).Infof("calling getter function of VmssFlexVMCache for vmss: %s", key)
+		localCache := &sync.Map{}
+
+		ctx, cancel := getContextWithCancel()
+		defer cancel()
+
+		vms, rerr := fs.VirtualMachinesClient.ListVmssFlexVMsWithoutInstanceView(ctx, key)
+		if rerr != nil {
+			klog.Errorf("ListVmssFlexVMsWithoutInstanceView failed: %v", rerr)
+			return nil, rerr.Error()
+		}
+
+		for i := range vms {
+			vm := vms[i]
+			if vm.Name != nil {
+				localCache.Store(*vm.Name, &vm)
+				fs.vmssFlexVMNameToVmssID.Store(*vm.Name, key)
+			}
+		}
+
+		return localCache, nil
+	}
+
+	if fs.Config.VmssFlexVMCacheTTLInSeconds == 0 {
+		fs.Config.VmssFlexVMCacheTTLInSeconds = consts.VmssFlexVMCacheTTLInSeconds
+	}
+	return azcache.NewTimedcache(time.Duration(fs.Config.VmssFlexVMCacheTTLInSeconds)*time.Second, getter)
+}
+
+func (fs *FlexScaleSet) newVmssFlexVMStatusCache() (*azcache.TimedCache, error) {
+	getter := func(key string) (interface{}, error) {
+		localCache := &sync.Map{}
+
+		ctx, cancel := getContextWithCancel()
+		defer cancel()
+
+		vms, rerr := fs.VirtualMachinesClient.ListVmssFlexVMsWithOnlyInstanceView(ctx, key)
+		if rerr != nil {
+			klog.Errorf("ListVmssFlexVMsWithOnlyInstanceView failed: %v", rerr)
+			return nil, rerr.Error()
+		}
+
+		for i := range vms {
+			vm := vms[i]
+			if vm.Name != nil {
+				localCache.Store(*vm.Name, &vm)
+				fs.vmssFlexVMNameToVmssID.Store(*vm.Name, key)
+			}
+		}
+
+		return localCache, nil
+	}
+
+	if fs.Config.VmssFlexVMStatusCacheTTLInSeconds == 0 {
+		fs.Config.VmssFlexVMStatusCacheTTLInSeconds = consts.VmssFlexVMStatusCacheTTLInSeconds
+	}
+	return azcache.NewTimedcache(time.Duration(fs.Config.VmssFlexVMStatusCacheTTLInSeconds)*time.Second, getter)
+}
+
+func (fs *FlexScaleSet) getNodeVmssFlexID(nodeName string) (string, error) {
+	klog.V(2).Infof("calling fs.getNodeVmssFlexID(%s)", nodeName)
+	fs.lockMap.LockEntry("getNodeVmssFlexID")
+	defer fs.lockMap.UnlockEntry("getNodeVmssFlexID")
+	vmssFlexID, isCached := fs.vmssFlexVMNameToVmssID.Load(nodeName)
+	if !isCached {
+		klog.V(2).Infof("nodeName %s is not saved in vmssFlexVMnameToVmssID map, send a GET request to retrieve its VmssID", nodeName)
+		machine, err := fs.getVirtualMachine(types.NodeName(nodeName), azcache.CacheReadTypeUnsafe)
+		if err != nil {
+			klog.Errorf("fs.getNodeVmssFlexID failed when Get VM: %v", err)
```

Comment:
it's not necessary to print error logs in every line since if it returns error, `getNodeVmssFlexID` would return error, the error could be printed in the caller of this function.

Reply:
Sure, just removed